cczoo/cross_lang_framework/clf_server/clf_server.c

```diff
@@ -89,37 +98,55 @@ int main(int argc, char** argv) {
 		return ret;
 
 	int len = 0;
-	char szVal[72];
+	int16_t tmp16 = 0;
+	char szVal[PATH_MAX] = {0};
 	const char conf[] = "clf_server.conf";
 	read_config(conf, "MRSigner", szVal, MRSIGNER_LEN, &len);
-	printf("MRSigner=%s\n", szVal);
+	printf("conf.MRSigner=%s\n", szVal);
 	hexstr2buff(szVal, g_mrsigner, MRSIGNER_LEN);
-	dump_buff(g_mrsigner, MRSIGNER_LEN);
 
 	read_config(conf, "MREnclave", szVal, MRENCLAVE_LEN, &len);
-	printf("MREnclave=%s\n", szVal);
+	printf("conf.MREnclave=%s\n", szVal);
 	hexstr2buff(szVal, g_mrenclave, MRENCLAVE_LEN);
-	dump_buff(g_mrenclave, MRENCLAVE_LEN);
 
 	read_config_short(conf, "isv_prod_id", (int16_t*)&g_isv_prod_id);
-	printf("isv_prod_id=%d\n", g_isv_prod_id);
+	printf("conf.isv_prod_id=%d\n", g_isv_prod_id);
 
 	read_config_short(conf, "isv_svn", (int16_t*)&g_isv_svn);
-	printf("isv_svn=%d\n", g_isv_svn);
+	printf("conf.isv_svn=%d\n", g_isv_svn);
 
 	read_config(conf, "secret", szVal, WRAP_KEY_SIZE, &len);
-	printf("secret=%s\n", szVal);
+	printf("conf.secret=%s\n", szVal);
 	strcpy(g_secret_pf_key_hex, szVal);
 
-	puts("Starting server on port 4433.");
-	/*TODO: make port/certification configurable */
+	read_config_short(conf, "port", (int16_t*)&tmp16);
+	if(tmp16 != 0)
+		g_port = tmp16;
```

Comment:
g_port is uint16, while tmp16 is int16, what the port is a negative number?.

Reply:
good catch!